netlink-packet/src/rtnl/tc/message.rs

```diff
@@ -0,0 +1,130 @@
+use super::{TcBuffer, TcNla};
+use failure::ResultExt;
+use {DecodeError, Emitable, Parseable, TC_HEADER_LEN};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct TcMessage {
+    header: TcHeader,
+    nlas: Vec<TcNla>,
+}
+
+impl TcMessage {
+    pub fn new() -> Self {
+        TcMessage::from_parts(TcHeader::new(), vec![])
+    }
+
+    pub fn into_parts(self) -> (TcHeader, Vec<TcNla>) {
+        (self.header, self.nlas)
+    }
+
+    pub fn header_mut(&mut self) -> &mut TcHeader {
+        &mut self.header
+    }
+
+    pub fn header(&self) -> &TcHeader {
+        &self.header
+    }
+
+    pub fn nlas(&self) -> &[TcNla] {
+        self.nlas.as_slice()
+    }
+
+    pub fn nlas_mut(&mut self) -> &mut Vec<TcNla> {
+        &mut self.nlas
+    }
+
+    pub fn append_nla(&mut self, nla: TcNla) {
+        self.nlas.push(nla)
+    }
+
+    pub fn from_parts(header: TcHeader, nlas: Vec<TcNla>) -> Self {
+        TcMessage { header, nlas }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct TcHeader {
+    family: u8,
+    // Interface index
+    index: i32,
+    // Qdisc handle
+    handle: u32,
+    // Parent Qdisc
+    parent: u32,
+    info: u32,
+}
+
+impl Default for TcHeader {
+    fn default() -> Self {
+        TcHeader::new()
+    }
+}
+
+impl TcHeader {
+    pub fn new() -> Self {
+        TcHeader {
+            family: 0,
+            index: 0,
+            handle: 0,
+            parent: 0,
+            info: 0,
+        }
+    }
+}
+
+impl Emitable for TcHeader {
+    fn buffer_len(&self) -> usize {
+        TC_HEADER_LEN
+    }
+
+    fn emit(&self, buffer: &mut [u8]) {
```

Comment:
Should we set the padding fields to `0` here? In theory they can be ignored but I don't know, it feels weird to have uninitialized bytes in the buffer.